Kickstarter-iOS/Views/Controllers/DiscoveryNavigationHeaderViewControllerTests.swift

```diff
@@ -19,12 +19,11 @@ internal final class DiscoveryNavigationHeaderViewControllerTests: TestCase {
 
   override func setUp() {
     super.setUp()
-    AppEnvironment.pushEnvironment(mainBundle: Bundle.framework)
```

Comment:
Just want to understand this change and the one on line 27. I think the pushing and popping of the environment may have been intentional in this test. I put these lines back and ran the tests locally and they fail because the debug menu icon appears on the header. I'm wondering if we'd rather want to fix that by not showing the debug icon unless we're running the debug build specifically?

One option is this line could change to `AppEnvironment.pushEnvironment(mainBundle: MockBundle(bundleIdentifier: KickstarterBundleIdentifier.test.rawValue))` now that we have the test bundle identifier and we can bring back the pushing and popping of environments?

But I think ideally we don't want to be making too many decisions based on whether we're in a test or not, so this might point to our booleans that we changed in `DiscoveryNavigationHeaderViewModel.swift`'s `debugContainerViewIsHidden` being the wrong approach.

Perhaps we should revert back to explicitly checking those booleans to show/hide the menu. Here's another suggestion:

1. Add something like this to `NSBundleType` (naming could be better):
```
public var isUserTestableBuild: Bool {
  return self.isBeta || self.isAlpha || self.isDebug
}
```
2. Change whether to show/hide the debug menu to:
```
self.debugContainerViewIsHidden = self.viewDidLoadProperty.signal
  .map { !AppEnvironment.current.mainBundle.isUserTestableBuild }
```
3. Remove `isTest` because we shouldn't need it in this case.

Lemme know what you think!

Reply:
Sounds good to me! Added a helper func to `DiscoveryNavigationHeaderViewModel` 👍